azure-pipelines.yaml

```diff
@@ -0,0 +1,61 @@
+name: Xamarin.Android.Tools $(Rev:r)
+variables:
+  major: 1
+  minor: 0
+  patch: $[counter('versioncounter', 100)]
```

Comment:
What is "patch", and how does it related to "build" and "revision"? https://docs.microsoft.com/en-us/dotnet/api/system.version.-ctor?view=netframework-4.7.2#System_Version__ctor_System_Int32_System_Int32_System_Int32_System_Int32_

*Does it* related to build & revision?  Or is it some NuGet-specific construct?

I'm also not sure what to make of the semantic "NuGet packages are the build number + 100"; presumably we'd wind up with an overlap at *some* point, right?  Unless we "regularly" change the major & minor values more frequently than once every 100 commits?

Assuming we *did* want to "'regularly' change the major & minor values," presumably this means the version numbers should match the Xamarin.Android release values, meaning [the version should be](https://github.com/xamarin/xamarin-android/blob/434d2d888633b11023dfee4cc7cc44ba88666b4e/Configuration.props#L15) v9.2?

I'm really not sure how to sanely handle the version numbers here. :-/

Reply:
I'm also not sure of the best approach with respect to versioning. The use of "patch" as a variable name here was only done with [Semantic Versioning](https://semver.org/) in mind, and would theoretically map to the `Build` property of `System.Version`.

As for version overlap, I believe the "patch" value will increment ~indefinitely using this approach, so we won't have to worry about changing the major/minor versions for the sole purpose of preventing version overlap.

As for keeping the version in sync with Xamarin.Android releases I don't feel that this should be a requirement, and I am not sure how desirable it is. Perhaps we'd be better served by only updating the major/minor versions alongside code changes in this repository, depending on the scope of those changes?